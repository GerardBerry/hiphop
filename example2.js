"use strict"

var reactive = require("./reactive-kernel.js");

/*
  <ReactiveMachine>
     <loop>
        <emit signal=${s} />
        <present signal=${s}>
           <emit signal=${t} />
        </present>
        <pause />
	<emit signal=${v} />
     </loop>
  </ReactiveMachine>
*/

var sigS = new reactive.Signal("S", false, function() {
   console.log("EMIT S");
});

var sigT = new reactive.Signal("T", false, function() {
   console.log("EMIT T");
});

var sigV = new reactive.Signal("V", false, function() {
   console.log("EMIT V");
});

var emitS = new reactive.Emit(sigS);
var emitT = new reactive.Emit(sigT);
var emitV = new reactive.Emit(sigV);
var present = new reactive.Present(sigS, emitT);
var pause = new reactive.Pause();
var seq = new reactive.Sequence(emitS, present, pause, emitV);
var loop = new reactive.Loop(seq);
var machine = new reactive.ReactiveMachine(loop);

for (var i = 0; i < 5; i++)
   machine.react(i);
