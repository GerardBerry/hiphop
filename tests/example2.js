"use hopscript"

var hh = require("hiphop");

var prg = <hh.module>
  <hh.outputsignal name="T"/>
  <hh.outputsignal name="V"/>
  <hh.localsignal name="S">
    <hh.loop>
      <hh.sequence>
	<hh.emit signal="S"/>
	<hh.present signal="S">
	  <hh.emit signal="T"/>
	</hh.present>
	<hh.pause/>
	<hh.emit signal="V"/>
      </hh.sequence>
    </hh.loop>
  </hh.localsignal>
</hh.module>

exports.prg = new hh.ReactiveMachine(prg, "example2");
