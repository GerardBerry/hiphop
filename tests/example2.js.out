---- reaction 0 GO:true RES:true INIT ----
EMIT S
EMIT T
---- SEL:true K0:false K1:true ----

---- reaction 1 GO:false RES:true ----
EMIT V
EMIT S
EMIT T
---- SEL:true K0:false K1:true ----

---- reaction 2 GO:false RES:true ----
EMIT V
EMIT S
EMIT T
---- SEL:true K0:false K1:true ----

---- reaction 3 GO:false RES:true ----
EMIT V
EMIT S
EMIT T
---- SEL:true K0:false K1:true ----

---- reaction 4 GO:false RES:true ----
EMIT V
EMIT S
EMIT T
---- SEL:true K0:false K1:true ----

